phoenix-core-client/src/main/java/org/apache/phoenix/expression/util/bson/SQLComparisonExpressionUtils.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.phoenix.expression.util.bson;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+import org.bson.RawBsonDocument;
+import org.mvel2.MVEL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * SQL style condition expression evaluation support.
+ */
+public class SQLComparisonExpressionUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SQLComparisonExpressionUtils.class);
+
+  private final RawBsonDocument rawBsonDocument;
+  private final BsonDocument comparisonValuesDocument;
+
+  public SQLComparisonExpressionUtils(RawBsonDocument rawBsonDocument,
+      BsonDocument comparisonValuesDocument) {
+    this.rawBsonDocument = rawBsonDocument;
+    this.comparisonValuesDocument = comparisonValuesDocument;
+  }
+
+  public boolean isConditionExpressionMatching(final String conditionExpression) {
+    String expression = convertExpression(conditionExpression);
+    LOGGER.info("Evaluating Expression: {}", expression);
+    try {
+      Object result = MVEL.eval(expression, this);
+      return result instanceof Boolean && (Boolean) result;
+    } catch (Exception e) {
+      LOGGER.error("Error while evaluating expression: {}", expression, e);
+      throw new RuntimeException("Expression could not be evaluated: " + expression);
+    }
+  }
+
+  public String convertExpression(String expression) {
+    expression =
+        expression.replaceAll("field_not_exists\\(([^)]+)\\)", "!exists('$1')");
+    expression = expression.replaceAll("field_exists\\(([^)]+)\\)", "exists('$1')");
+    //    expression = expression.replaceAll("\\b([\\w.\\[\\]]+)\\b\\s*=\\s*(:\\w+)", "isEquals('$1', '$2')");
+    expression =
+        expression.replaceAll("\\b([\\w.\\[\\]]+)\\s*=\\s*([#:$]*\\w+)", "isEquals('$1', '$2')");
```

Comment:
these regex should be declared as some constants?

Reply:
done